de.dentrassi.pm.aspect.mvnosgi/src/de/dentrassi/pm/aspect/mvnosgi/internal/VirtualizerImpl.java

```diff
@@ -43,15 +42,41 @@ public void virtualize ( final Context context )
 
     private void processVirtualize ( final Context context ) throws Exception
     {
-        final ArtifactInformation art = context.getArtifactInformation ();
+        // The group ID will be set using the provided channel meta data or the default one.
+        String groupId = null;
+
+        // Extract some informations (e.g. group ID) from the provided channel meta data.
+        final Map<MetaKey, String> channelMetaData = context.getProvidedChannelMetaData ();
+        for ( final Entry<MetaKey, String> entry : channelMetaData.entrySet () )
```

Comment:
Is this just for information or would you like this changed?

At the moment there is only one channel meta data present and used, but this could be changed.
Using this implementation the look up (independ of the number of parameter) is always O(n) (for the whole evaluation).
Using a direct get on the map need O(log(n)) for every access (could result in more overhead if we are using multiple ones).